security/pkg/pki/ra/k8s_ra.go

```diff
@@ -77,16 +77,22 @@ func (r *KubernetesRA) Sign(csrPEM []byte, certOpts ca.CertOpts) ([]byte, error)
 }
 
 // SignWithCertChain is similar to Sign but returns the leaf cert and the entire cert chain.
-func (r *KubernetesRA) SignWithCertChain(csrPEM []byte, certOpts ca.CertOpts) ([]byte, error) {
+func (r *KubernetesRA) SignWithCertChain(csrPEM []byte, certOpts ca.CertOpts) ([]string, error) {
+	certChain := []string{}
 	cert, err := r.Sign(csrPEM, certOpts)
 	if err != nil {
-		return nil, err
+		return certChain, err
 	}
+	certChain = append(certChain, string(cert))
 	chainPem := r.GetCAKeyCertBundle().GetCertChainPem()
 	if len(chainPem) > 0 {
-		cert = append(cert, chainPem...)
+		certChain = append(certChain, string(chainPem))
+	}
+	rootPEM := r.GetCAKeyCertBundle().GetRootCertPem()
```

Comment:
Although the root can be added in the cert chain, I still don't understand how this solves the issue? The solution proposed in https://github.com/istio/istio/issues/32403 doesn't work?

Reply:
The main goal of this set of commits was to *remove* checking of the Ca certificate completely when Istio is being used as an RA to sign workload certificates (not for the Istiod certificate itself). This is not needed IMO. The next commit will handle this.